src/containers/home/home.component.jsx

```diff
@@ -8,19 +8,46 @@ import Navbar from "containers/navbar";
 
 const Home = (props) => {
   const [posts, setPosts] = useState([]);
-
+  const [displayPosts, setDisplayPosts] = useState([]);
+  const [selectDisplayPostOption, setSelectDisplayPostOption] = useState(
+    "Newest"
+  );
+  const [viewByTimeframeTime, setViewByTimeframeTime] = useState(Date.now());
+  const [viewByTimeframe, setViewByTimeframe] = useState("All-time");
   const userFromState = useSelector((state) => state.users.user);
+  const userVoteHistory = userFromState.vote_history;
   const doesUserHaveUsername = !!userFromState.username;
+  const postsFromState = useSelector((state) => state.posts);
+
   const signOut = () => {
     auth.signOut();
     props.logoutThunk();
   };
-  const postsFromState = useSelector((state) => state.posts);
+
+  const handleDisplayPostChange = () => {
+    const viewOption = document.getElementById("select-view-option").value;
+    setSelectDisplayPostOption(viewOption);
+    const timeframe = document.getElementById("select-view-option-timeline")
+      .value;
+    setViewByTimeframe(timeframe);
+    if (timeframe === "All-time") {
+      setViewByTimeframeTime(Date.now());
+    } else if (timeframe === "Today") {
+      setViewByTimeframeTime(86400000);
+    } else if (timeframe === "This week") {
+      setViewByTimeframeTime(604800000);
+    }
+  };
 
   useEffect(() => {
-    props.fetchPostsThunk();
+    console.log(selectDisplayPostOption);
+    if (selectDisplayPostOption === "Newest") {
+      props.fetchPostsThunk("newest");
```

Comment:
Can we chance `fetchPostsThunk` to accept an object as an argument so that we are clear what we are passing in?

Reply:
Right, it wasnt going to work one way, but might now that i've tweaked it
Ill try

Reply:
done, going to make that function more modular